### PR TITLE
gif-for-cli: init and pythonPackages.x256: init at 0.0.3

### PR DESCRIPTION
**Pull request by Scriptkiddi**
###### Motivation for this change
Wanted to use it for my dotfiles

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ x Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by dotlambda — CHANGES_REQUESTED** [review]
PRs should be made against master: https://nixos.org/nixpkgs/manual/#submitting-changes-making-patches

**Review by dotlambda — CHANGES_REQUESTED** [review]
Please create 2 separate commits.

File: pkgs/tools/misc/gif-for-cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{ stdenv, fetchFromGitHub, python3Packages, ffmpeg, zlib, libjpeg }:
+
+python3Packages.buildPythonApplication rec {
+  name = "gif-for-cli";
```
[the remark this answers is not in the document]
Comment (Scriptkiddi) [reply pkgs/tools/misc/gif-for-cli/default.nix]: i will add pname, 
gif-for-cli has no version number which should I take then

File: pkgs/tools/misc/gif-for-cli/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    homepage = https://github.com/google/gif-for-cli;
+    description = "Takes in a GIF, short video, or a query to the Tenor GIF API and converts it to animated ASCII art.";
+    license = licenses.apache2;
+    maintainers = with maintainers; [ scriptkiddi ];
+    platforms = platforms.linux;
+  };
+
+  checkInputs = [ python3Packages.coverage ];
+
+  src = fetchFromGitHub {
```
[the remark this answers is not in the document]
Comment (Scriptkiddi) [reply pkgs/tools/misc/gif-for-cli/default.nix]: it does not have a pypi package

File: pkgs/tools/misc/gif-for-cli/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (dotlambda) [reply pkgs/tools/misc/gif-for-cli/default.nix]: Then use `unstable-YYYY-MM-DD` with the date of the commit you use.

**Comment by Scriptkiddi** [comment]
switeched over here https://github.com/NixOS/nixpkgs/pull/52201

**Pull request closed without merging**